components/camel-braintree/src/main/java/org/apache/camel/component/braintree/BraintreeConfiguration.java

```diff
@@ -104,7 +108,7 @@ public void setMethodName(String methodName) {
     }
 
     public String getEnvironment() {
-        return ObjectHelper.notNull(environment, ENVIRONMENT);
+        return environment;
```

Comment:
or my understanding, why ObjectHelper.notNull has been removed ?

Reply:
The reason for this is because when using an Access Token, the environment option is no longer required.  The same thing was done for the merchantId, publicKey and privateKey options.

I would have liked to have something like "if configured using public and private keys, require environment, merchantId, publicKey and privateKey, and if configured using an access token, require only the access token" but I couldn't figure out a way to do that.

Reply:
Ok understood. It looks good for me but you may need to amend also the [ServiceNowComponentVerifierExtension](https://github.com/apache/camel/blob/master/components/camel-servicenow/camel-servicenow-component/src/main/java/org/apache/camel/component/servicenow/ServiceNowComponentVerifierExtension.java#L39-L47) and add conditional validation  according to what is given as properties

Reply:
The ServiceNow component is new to me, and I am not sure where I would even start. Is there any other component with conditional validation I might use as a model?

Reply:
[This](https://github.com/apache/camel/blob/master/components/camel-salesforce/camel-salesforce-component/src/main/java/org/apache/camel/component/salesforce/SalesforceComponentVerifierExtension.java#L43-L67) may help you , if you need any assistence ping me on irc

Reply:
I attempted setting up a VerifierExtension - I think it's ok, but feedback would be welcome.

Also, if the pull request needs to be squashed to one commit, let me know.

Reply:
@nofuss can you please squash in one commit and add a reference to the JIRA number in the commit message? Thanks.

Reply:
done